src/main/java/jenkins/plugins/nodejs/tools/NpmPackagesBuildWrapper.java

```diff
@@ -1,112 +1,50 @@
-        return new Environment(){
-            @Override
-            public boolean tearDown(AbstractBuild build, BuildListener listener)
-                    throws IOException, InterruptedException {
-                return true;
-            }
-        };
-    }
-
     public String getNodeJSInstallationName() {
         return nodeJSInstallationName;
     }
 
     @Override
-    public Launcher decorateLauncher(final AbstractBuild build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException, Run.RunnerAbortedException {
-        return new DecoratedLauncher(launcher){
-            @Override
-            public Proc launch(ProcStarter starter) throws IOException {
-                // Avoiding potential NPE when calling starter.envs()
-                // Yes, this is weird...
-                String[] starterEnvs;
-                try {
-                   starterEnvs = starter.envs();
-                } catch (NullPointerException ex) {
-                    starterEnvs = new String[0];
-                }
-
-                String pathSeparator = File.pathSeparator;
-
-                EnvVars vars = toEnvVars(starterEnvs);
-
-                NodeJSInstallation nodeJSInstallation = 
-                    NodeJSPlugin.instance().findInstallationByName(nodeJSInstallationName);
-
-                try {
-                    nodeJSInstallation = nodeJSInstallation.forNode(build.getBuiltOn(), listener);
-                    nodeJSInstallation = nodeJSInstallation.forEnvironment(vars);
-
-                    Computer slave = Computer.currentComputer();
-                    String slavePathSeparator = (String)slave.getSystemProperties().get("path.separator");
-
-                    if (slavePathSeparator != null) {
-                        pathSeparator = slavePathSeparator;
-                    }
-                } catch (InterruptedException e) {
-                    Throwables.propagate(e);
-                }
-
-                // HACK: Avoids issue with invalid separators in EnvVars::override in case of different master/slave
-                
-                String overriddenPaths = NodeJSInstaller.binFolderOf(nodeJSInstallation, build.getBuiltOn())
-                        + pathSeparator
-                        + vars.get("PATH");
-                vars.override("PATH", overriddenPaths);
-
-                return super.launch(starter.envs(Util.mapToEnv(vars)));
-            }
-
-            private EnvVars toEnvVars(String[] envs) {
-                EnvVars vars = new EnvVars();
-                for (String line : envs) {
-                    vars.addLine(line);
-                }
-                return vars;
-            }
-        };
+    public void setUp(Context context, Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener, EnvVars env) throws IOException, InterruptedException {
+
+        final Computer computer = workspace.toComputer();
```

Comment:
FilePath.toComputer can return null, but does this make any sense for a workspace ?
i.e. CheckForNull is not context aware